### PR TITLE
Upgrade System.IO csproj files to target newer .NET version

### LINKED ISSUE #7086: Upgrade from `netcoreapp2.1` to LTS (or current) version
The following project file(s) target a .NET version which is no longer supported. This is an auto-generated issue, detailed and discussed in [dotnet/docs#22271](https://github.com/dotnet/docs/issues/22271).

| Target version | End of life | Release notes | Nearest LTS TFM version |
| --- | --- | --- | --- |
| `netcoreapp2.1` | August, 21 2021 | [netcoreapp2.1 release notes](https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/2.1/releases.json) | `netcoreapp3.1` |

  - [x] [samples/snippets/csharp/api/system.io/path/combine/misc.csproj](../blob/main/samples/snippets/csharp/api/system.io/path/combine/misc.csproj#L5)
  - [x] [samples/snippets/csharp/api/system.io/path/combine2/combine.csproj](../blob/main/samples/snippets/csharp/api/system.io/path/combine2/combine.csproj#L5)
  - [x] [samples/snippets/csharp/api/system.io/path/combine3/combine.csproj](../blob/main/samples/snippets/csharp/api/system.io/path/combine3/combine.csproj#L5)

Consider upgrading projects to either the current release, or the nearest LTS TFM version.

If any of these projects are intentionally targeting an unsupported version, you can optionally configure to ignore including them in this automated issue. Create a file at the root of the repository, named *dotnet-versionsweeper.json* and add an `ignore` entry following the [globbing patterns detailed here](https://docs.microsoft.com/dotnet/api/microsoft.extensions.filesystemglobbing.matcher#remarks).

```json
{
    "ignore": [
        "**/path/to/example.csproj"
    ]
}
```

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Upgrade from `netcoreapp2.1` to `netcoreapp3.1`

Fixes #7086

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
LGTM

File: samples/snippets/csharp/api/system.io/path/combine/misc.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (carlossanlop) [inline samples/snippets/csharp/api/system.io/path/combine/misc.csproj:5]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>

```

File: samples/snippets/csharp/api/system.io/path/combine2/combine.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (carlossanlop) [inline samples/snippets/csharp/api/system.io/path/combine2/combine.csproj:5]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>

```

File: samples/snippets/csharp/api/system.io/path/combine3/combine.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (carlossanlop) [inline samples/snippets/csharp/api/system.io/path/combine3/combine.csproj:5]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>

```

**Review by carlossanlop — COMMENTED** [review]
Let's just go ahead and target .NET 6.0. It works. Example: https://github.com/dotnet/dotnet-api-docs/pull/7267

**Pull request merged**